query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -228,6 +228,53 @@ class QuantileOverTimeFunction(funcParams: Seq[Any]) extends RangeFunction {
   }
 }
 
+class MedianAbsoluteDeviationOverTimeFunction(funcParams: Seq[Any]) extends RangeFunction {
+  override def addedToWindow(row: TransientRow, window: Window): Unit = {
+  }
+
+  override def removedFromWindow(row: TransientRow, window: Window): Unit = {
+  }
+
+  override def apply(
+                      startTimestamp: Long, endTimestamp: Long, window: Window,
+                      sampleToEmit: TransientRow,
+                      queryConfig: QueryConfig
+                    ): Unit = {
+    val q = 0.5
+    val values: Buffer[Double] = Buffer.ofSize(window.size)
+    var i = 0
+    while (i < window.size) {
+      val curValue = window.apply(i).getDouble(1)
+      if (!curValue.isNaN) {
+        values.append(curValue)
+      }
+      i = i + 1
+    }
+    val counter = values.length
```

Comment:
Could you rename this to something like `length`? Or remove `counter` and reference `values.length` directly?

Reply:
Might even be easier to handle `values.length == 0` once at the beginning of the method -- what do you think?